app/assets/javascripts/discourse/app/controllers/sidebar-section-form.js

```diff
@@ -0,0 +1,228 @@
+    );
+  }
+
+  get valueCssClass() {
+    return this.value === undefined || this.validValue ? "" : "warning";
+  }
+}
+
+export default Controller.extend(ModalFunctionality, {
+  dialog: service(),
+  router: service(),
+
+  onShow() {
+    this.setProperties({
+      flashText: null,
+      flashClass: null,
+    });
+    this.model = this.initModel();
+  },
+
+  onClose() {
+    this.model = null;
+  },
+
+  initModel() {
+    if (this.model) {
+      return new Section({
+        title: this.model.title,
+        links: A(
+          this.model.links.map(
+            (link) =>
+              new SectionLink({
+                router: this.router,
+                name: link.name,
+                value: link.value,
+                id: link.id,
+              })
+          )
+        ),
+        id: this.model.id,
+      });
+    } else {
+      return new Section({
+        links: A([new SectionLink({ router: this.router })]),
+      });
+    }
+  },
+
+  create() {
+    return ajax(`/sidebar_sections/`, {
+      type: "POST",
+      data: {
+        title: this.model.title,
+        links: this.model.links.map((link) => {
+          return {
+            id: link.id,
+            name: link.name,
+            value: link.path,
+          };
+        }),
+      },
+    })
+      .then((data) => {
+        this.currentUser.sidebar_sections.pushObject(data.sidebar_section);
+        this.send("closeModal");
+      })
+      .catch((e) =>
+        this.setProperties({
+          flashText: sanitize(extractError(e)),
+          flashClass: "error",
+        })
+      );
+  },
+
+  update() {
+    return ajax(`/sidebar_sections/${this.model.id}`, {
+      type: "PUT",
+      data: {
+        title: this.model.title,
+        links: this.model.links.map((link) => {
+          return {
+            id: link.id,
+            name: link.name,
+            value: link.path,
+            _destroy: link._destroy,
+          };
+        }),
+      },
+    })
+      .then((data) => {
+        const newSidebarSections = this.currentUser.sidebar_sections.map(
+          (section) => {
+            if (section.id === data["sidebar_section"].id) {
+              return data["sidebar_section"];
+            }
+            return section;
+          }
+        );
+        this.currentUser.set("sidebar_sections", newSidebarSections);
+        this.send("closeModal");
+      })
+      .catch((e) =>
+        this.setProperties({
+          flashText: sanitize(extractError(e)),
+          flashClass: "error",
+        })
+      );
+  },
+
+  get activeLinks() {
+    return this.model.links.filter((link) => !link._destroy);
+  },
+
+  actions: {
+    addLink() {
+      this.model.links.pushObject(new SectionLink({ router: this.router }));
+    },
+
+    deleteLink(link) {
+      this.model.links.removeObject(link);
+      if (link.id) {
+        link._destroy = "1";
+        this.model.links.pushObject(link);
+      }
```

Comment:
The logic here feels confusing to me. If a link has an `id` attribute, why don't we just update the `_destroy` attribute for the link instead of removing the object, updating the attribute and pushing the object back into `this.model.links`.